### PR TITLE
Docs/less hierarchy reqs

### PR DESCRIPTION
**Pull request by cbkerr**
## Description

I tried to make the section on install requirements a bit easier to skim by getting rid of the all-encompassing top level heading.
## How has this been tested?
I've built the docs locally and made sure the same list structure appears

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: INSTALLING.rst [hunk 1]
```diff
@@ -91,31 +91,28 @@ Tell CMake to search for packages in the virtual environment first::
...
-  - pybind11 >= 2.2
-  - Eigen >= 3.2
-  - CMake >= 3.9
-  - For MPI parallel execution (required when ``ENABLE_MPI=on``):
-
+**HOOMD-blue** requires a number of libraries to build. The flags ``ENABLE_MPI``,
+``ENABLE_GPU``, ``ENABLE_TBB``, and ``BUILD_JIT`` each require additional libraries.
+
+General requirements
+    - C++11 capable compiler (tested with ``gcc`` 4.8, 5.5, 6.4, 7,
```
Comment (joaander) [inline INSTALLING.rst:98]: These '-''s need to be in the first column. There also needs to be a blank line between `General requirements` and the start of the list. 
```suggestion

- C++11 capable compiler (tested with ``gcc`` 4.8, 5.5, 6.4, 7,
```

File: INSTALLING.rst [hunk 2]
```diff
@@ -91,31 +91,28 @@ Tell CMake to search for packages in the virtual environment first::
...
+      8, 9, ``clang`` 5, 6, 7, 8)
+    - Python >= 3.5
+    - NumPy >= 1.7
+    - pybind11 >= 2.2
+    - Eigen >= 3.2
+    - CMake >= 3.9
+
+For MPI parallel execution (required when ``ENABLE_MPI=on``):
     - MPI (tested with OpenMPI, MVAPICH)
```
Comment (joaander) [inline INSTALLING.rst:107]: Same here:
```suggestion

- MPI (tested with OpenMPI, MVAPICH)
```

File: INSTALLING.rst [hunk 3]
```diff
@@ -91,31 +91,28 @@ Tell CMake to search for packages in the virtual environment first::
...
     - NVIDIA CUDA Toolkit >= 9.0
 
-    **OR**
-
-    - `AMD ROCm >= 2.9 <https://rocm.github.io/ROCmInstall.html>`_
+    *OR*
 
-      Additional dependencies:
+    - `AMD ROCm >= 2.9 <https://rocm.github.io/ROCmInstall.html>`_ with additional dependencies:
         - HIP [with ``hipcc`` and ``hcc`` as backend]
```
Comment (joaander) [inline INSTALLING.rst:116]: In nested lists, there needs to be a blank line before the start *and* the `-` needs to line up with the first character of the parent list text (the backtick above).

File: INSTALLING.rst [hunk 4]
```diff
@@ -131,21 +128,16 @@ Install prerequisites
          A `temporary bugfix branch of HIP <https://github.com/glotzerlab/HIP/tree/hipfuncgetattributes_revertvectortypes>`_
```
Comment (joaander) [inline INSTALLING.rst:128]: The block starting with `A temporary bugfix branch` no longer applies and should be removed. The ROCm requirement should be increased to >= 3.5.0. (#750).

**Review by joaander — CHANGES_REQUESTED** [review]
Thanks, this cleans things up nicely.

Can you fix the list formatting so that sphinx doesn't render the headings with blue banners? Those are definition lists and we want just a normal bulleted list.

Restructured text is *very* particular about list formatting: https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#bullet-lists

I'll note a few of the needed changes below.

**Comment by cbkerr** [comment]
I think I've smoothed out most of the formatting issues.

**Review by joaander — APPROVED** [review]
Thanks!

**Pull request merged**